### ISSUE TITLE
Negative "Non-fossil energy consumed" when manually charging battery pack

### ISSUE THREAD

**Issue by g1za** [home-assistant/frontend#18656]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

I'm charging my battery pack at night when the cost is the lowest.
I noticed that the guauge shows negative percentages when this is done; maybe this presumes you can charge your battery only when there is PV production?
Percentage progressively go back to something normal as the day progresses.

### Describe the behavior you expected

I would expect that the energy used to charge the battery is not accounted when there is no PV production as (negative) "Non-fossil energy". It should just be zero.

### Steps to reproduce the issue

1. Manually charge the batteries from the grid at night (I assume any time there is no PV production)
2. Check the energy dashboard "Non-fossil energy consumed" gauge during the day

![Screenshot 2023-11-14 alle 07 35 51](https://github.com/home-assistant/frontend/assets/88257868/287588aa-9e55-4237-ac28-4932933ede2d)

![Screenshot 2023-11-14 alle 14 25 17](https://github.com/home-assistant/frontend/assets/88257868/818794d3-d9dd-4832-a699-2e5b35b9293a)

### What version of Home Assistant Core has the issue?

2023.11.2

### In which browser are you experiencing the issue with?

Tested with Firefox 119.0.1 and mobile companion app. Not a browser issue.

### Which operating system are you using to run this browser?

macOS Ventura 13.6.1 and iOS 7.1. Not an OS issue

**Comment by g1za** [comment]
It really looks like HA takes the assumption battery charge can only happen when there is PV production. 

![IMG_3456](https://github.com/home-assistant/frontend/assets/88257868/18856b6f-619a-4780-912a-61ae3c372b46)
![IMG_3457](https://github.com/home-assistant/frontend/assets/88257868/9a1bcbd9-d213-47f3-afc5-1ee27e7e09c9)
![IMG_3458](https://github.com/home-assistant/frontend/assets/88257868/536b2a62-524d-42ca-95e9-63a67ce2607f)
![IMG_3459](https://github.com/home-assistant/frontend/assets/88257868/e1f0365e-a3af-4272-a2b6-9755eb59fe02)

**Comment by g1za** [comment]
Issue still persistent

**Comment by lfarrand** [comment]
This issue is also affecting me.

I charge my batteries overnight at a cheaper rate and then discharge later in the day. I'd expect the following graph to show that the battery charging has consumed power, not produced it. At the moment it looks like the batteries are providing negative usage.

![image](https://github.com/home-assistant/frontend/assets/1841445/9ddecc87-7460-41ce-8074-1baed7fca7db)

In the above chart I wouldn't expect any negative values and would expect the red bars to be positive.

**Comment by g1za** [comment]
Tue bug is still there

**Comment by shanetuohy** [comment]
I've noticed this issue also and I believe it makes the energy dashboard less useful for those with batteries. 

I believe g1za is correct that the energy dashboard assumes that your batteries only charge from solar energy. This is not true for many setups, and lots of people avail of cheap energy at off peak times to charge their batteries.

**Comment by LizardWizardGB** [comment]
I have the same issue, with my Growatt battery.

**Comment by AndyMillne** [comment]
Same issue

<img width="1520" alt="Image" src="https://github.com/user-attachments/assets/c8479ca0-c83f-4df4-8c0e-5946ffdb508e" />

**Comment by linuxstony** [comment]
Would be nice if the devs could assign someone :) I am a subscriber and this value is wrong for me as well with a Zendure Hyper2000 and battery. Please correct the algorithm.

Thanks in advance.

**Comment by jamesloker** [comment]
I have a same issue.
Yesterday
![Image](https://github.com/user-attachments/assets/4f32a9b7-db6d-49f1-817f-59a3ad26cf3e)
Today
![Image](https://github.com/user-attachments/assets/e156949b-7735-4091-a175-4904a2e473d9)
![Image](https://github.com/user-attachments/assets/d197d6e5-20ea-445b-a015-9b86e96f6b75)

I noticed it started happening when i created a helper utility meter to replace the Enphase Envoy daily production entity. The enphase was showing a spike every morning at between midnight and 1am which caused the solar readings to be incorrect as per https://community.home-assistant.io/t/enphase-envoy-integration-showing-incorrect-daily-production/688033/2

Since then it shows the negative self sufficiency.

**Comment by jomach** [comment]
Same issue 
![image](https://github.com/user-attachments/assets/d33e2e89-c91c-44c1-9b21-17bd9798065d)
![image](https://github.com/user-attachments/assets/a273750c-613d-49e9-a07e-6fed347824a3)
![image](https://github.com/user-attachments/assets/3c1cd589-f6e7-4b7b-94e5-dd7f8723acf4)

**Comment by jomach** [comment]
Hi @karwosts sorry to Ping you but I saw comment the last issue related. Do you know someone who can help here?

**Comment by karwosts** [comment]
We had a big update of energy calculations in 2025.5. Can someone try on that version and see how it looks?

**Comment by jomach** [comment]
Same here. 
[Energy – Home Assistant.pdf](https://github.com/user-attachments/files/20117273/Energy.Home.Assistant.pdf)

**Comment by karwosts** [comment]
> Hi, yesterday after the update my self sufficiency was zero - despite solar input - i am fairly sure that this is wrong.

The self-sufficiency equation roughly says:

1 - (energyFromGrid / homeConsumption)

If your energy from grid is _greater_ than your total consumption, then that's why that would be zero (as it's clamped to not go below 0). 

If your consumption was 10kW and you achieved that with 5kW of input, that would score 50%. If you consumed 10kW with _zero_ grid input, that would be 100%. 

I'm not exactly sure what number you would like to see in this case.


> Same here.

@jomach this pdf simply doesn't sense to me. It says you have 11.63 of energy inputs (solar/grid_in/battery_out), and from that you managed to export/store 13.1 kWh (to_grid and to_battery). That doesn't add up?

**Comment by jomach** [comment]
That's the problem for me, I will wait for 1 day. Just upgrade to 2025.5 to see if it helps.

**Comment by karwosts** [comment]
Just to note nothing changed in 2025.5 on the _collection_ side, so looking at any of your history from previous days would be equally valid. 

Also why are we all discussing self sufficiency if this thread is titled about non-fossil energy gauge. Was that always just a typo?😕

**Comment by jomach** [comment]
So I just checked the data against my Victron UI and there is definitely something wrong. I think the main problem is that my battery is only one phase instead of three phases. My energy meter compensates the different basically. This is normal in Germany.
Victron data: 

![image](https://github.com/user-attachments/assets/4d105451-5c12-4be6-b11e-49663f3e6c5f)

![image](https://github.com/user-attachments/assets/c49eac0a-8221-4121-b9f5-9fd51f859cbb)

**Comment by g1za** [comment]
@karwosts no, that was not a typo. The problem I reported here is about negative non-fossil energy, as well documented. Everything about self consumption should be moved to another thread and maybe marked as spam in the current context.
About the bug I reported I need to manually charge my batteries at night, which may be tricky as during summer time they are already fully charge during the day. I let you know. 
Cheers!

**Comment by jomach** [comment]
@karwosts I will report a new ticket with the detailed explanation.

**Comment by karwosts** [comment]
The problem I reported here is about negative non-fossil energy, as well documented. 

But this is what you highlighted in your original post? That is self-sufficiency gauge?

![Image](https://github.com/user-attachments/assets/22895980-0be0-41ca-8c25-0f6a748e378a)

**Comment by g1za** [comment]
It is. But in my analysis that is an effect of the calculation of negative energy consumed when charging the batteries at night.

**Comment by jomach** [comment]
I don't charge at night and it is also wrong.

**Comment by jomach** [comment]
I don't charge at night and it is also wrong.

**Comment by g1za** [comment]
@jomach That’s why you should report in a different issue.

**Comment by karwosts** [comment]
@g1za - please update to 2025.5 and provide a new analysis if you still have an issue. I'm still not entirely sure I understand what point you're trying to make, but I believe support for charging batteries from grid and discharging batteries to grid should be greater supported now.

**Comment by g1za** [comment]
Thanks @karwosts 
I’m saying that charging batteries from grid is (or was) not supported (I believe the system assumed batteries could just be charged via solar panels and not grid, hence negative values). 
What I see is being reported is negative values, no matter what, that is not the original issue I reported

You’re right, I made a bit of confusion in my previous post, but let’s keep in mind I reported this over 1,5 years ago. 
At the moment I also have a problem with my internet provider and using his modem, so my local network is temporarily broken. 
I will test this as soon as possible and let you know. But probably not before two weeks also because of some business trips. 

Cheers

**Comment by jomach** [comment]
I created an issue for the Self-sufficiency here https://github.com/home-assistant/frontend/issues/25446
